### ISSUE TITLE
Consider if we should allow to create multiple publisher instances

### ISSUE THREAD

**Issue by KacperKluka** [ably/ably-asset-tracking-android#464]
After solving the issue about having a single Mapbox instance for the whole app (https://github.com/ably/ably-asset-tracking-android/issues/463) we might want to reconsider whether we want to allow the SDK users to create multiple publishers.

The reason behind this is that since we have a single instance of Mapbox then it would need to handle all the publishers. When creating a publisher we can specify some additional configurations for the Mapbox object such as the location source. Because we have a single instance of Mapbox then only the first publisher will be able to configure it. All the publishers created afterwards will be able to specify a Mapbox configuration but it won't have any effect since the Mapbox is already running.

Another issue is: when should we stop the Mapbox instance if there are multiple publishers? We should somehow know how many active publishers there are and only if the last one of them stops the Mapbox should be stopped.

┆Issue is synchronized with this [Jira Task](https://ably.atlassian.net/browse/SDK-1273) by [Unito](https://www.unito.io)

**Comment by KacperKluka** [comment]
On today's call, we've come up with another solution that sounds very promising. It involves extracting the Mapbox code to a separate object in our API (something like `GpsProvider`) which then will be used as a parameter of the publisher builder. This will make it clear to the users that the Mapbox instance is shared between publishers. We can additionally add some options to it like `auto-close` which will determine if the instance should be closed with the last publisher or if it should be retained for later use. We could even go as far as allowing the users to create their own implementation of the GPS provider API that might be useful in some rare cases :thinking: 
WDYT @AndyNicks @QuintinWillison

**Comment by mikelee638** [comment]
In discussion, we decided the next step here is to document the existing behavior for multiple publishers, and likely recommend that multiple publishers not be created since it will not work as developers intend.

We also discussed the comment above (https://github.com/ably/ably-asset-tracking-android/issues/464#issuecomment-941046561), and agree this would be a great enhancement (who doesn't love decoupling things). This work is captured in a separate issue (#718).
